sympy/solvers/solveset.py

```diff
@@ -339,10 +339,21 @@ def _invert_complex(f, g_ys, symbol):
                                imageset(Lambda(n, f.inverse()(n)), g_ys), symbol)
 
     if isinstance(f, exp):
-        if isinstance(g_ys, FiniteSet):
+        from sympy.abc import k,n
```

Comment:
What if the user has already used these symbols in their equation?

This should use `Dummy`

Reply:
So every arbitrary constant should be a Dummy!